Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -118,8 +119,76 @@ private static void GenerateRGScripts()
             }
         }
 
+        private static List<Scene> GetDirtyScenes()
+        {
+            var result = new List<Scene>();
+            for (int j = 0; j < SceneManager.sceneCount; j++)
+            {
+                var scene = SceneManager.GetSceneAt(j);
+                if (scene.isDirty)
+                {
+                    result.Add(scene);
+                }
+            }
+
+            return result;
+        }
+
         [MenuItem("Regression Games/Agent Builder/Extract Game Context")]
         private static void ExtractGameContext()
+        {
+            if (EditorUtility.DisplayDialog(
+                    "Extract Game Context",
+                    "Warning: This operation will load and unload every scene in your project's build configuration while gathering data.  This operation can take a long time to complete depending on the size of your project.",
+                    "Continue",
+                    "Cancel"))
+            {
+                var dirtyScenes = GetDirtyScenes();
+                if (dirtyScenes.Count > 0)
+                {
+                    if (EditorUtility.DisplayDialog(
+                            "Unsaved Changes Detected",
+                            "One or more open Scenes have unsaved changes.",
+                            "Save Scenes and Continue Extract",
+                            "Cancel"
+                        ))
+                    {
+                        foreach (var dirtyScene in dirtyScenes)
+                        {
+                            EditorSceneManager.SaveScenes(dirtyScenes.ToArray());
+                            ExtractGameContextHelper();
```

Comment:
Do we need to do call `ExtractGameContextHelper` for every scene, or can we do this once after all scenes are saved?

Reply:
yes, we need to call it for every scene

I start by calling for whatever (scene/scenes) they have open, then call it one by one for all other scenes as we open and close them

then.. restore their original scene state as closely as we can (ordering gets slightly changed if they had unloaded scenes in their view)

Reply:
oh... haha.. i forgot to take out the loop.. fixing

Reply:
fixed